### PR TITLE
Add code coverage comments for PR

### LINKED ISSUE #332: Add code coverage report in PR comments
It will be useful to have test code coverage report(with delta) available on every `PR` to help PR reviewers.  `codecov` provides such integration and adds a detailed report in PR comments - https://docs.codecov.com/docs/pull-request-comments

### PR DESCRIPTION
**Pull request by dharmjit**
**What this PR does / why we need it**:
This PR integrates the `codecov` comment in the PR conversation which will help reviewers to understand the changes in test code coverage. This PR includes
- Refactored `GH Actions` workflows for  agent, webhook, and controller and are merged now.
- `README.md` is updated to reflect CI workflow status on main branch
- `test`, `apis` folders are excluded to match with the code coverage report.

**Which issue(s) this PR fixes** :
Fixes #332 

**Additional Information**:
Test run and sample output is available here - https://github.com/dharmjit/cluster-api-provider-bringyourownhost/pull/8

### CODE REVIEW

**Review by jamiemonserrate — APPROVED** [review]
I took a quick look, and it seems fine to me. Took me a while to realise that we are now depending on `test-coverage` to run all our tests. 

Question - shouldn't the codecov file have been uploaded as part of this PR? Or will it work on subsequent PRs?

File: codecov.yml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+comment:                  # this is a top-level key
+  layout: "reach, diff, flags, files"
+  behavior: default
+  require_changes: false  # if true: only post the comment if coverage changes
```
Comment (anusha94) [inline codecov.yml:4]: Should we set this to true? No strong opinions but kinda feels unnecessary if there is no coverage change

**Comment by dharmjit** [comment]
> Took me a while to realise that we are now depending on `test-coverage` to run all our tests

@jamiemonserrate, are you suggesting to define another `make` target for GH Actions workflows. I assumed we used `make test-<component>` targets for local dev/test.

> Question - shouldn't the codecov file have been uploaded as part of this PR? Or will it work on subsequent PRs?

If you meant the codecov PR comment, somehow it was not there on the first CI run, I reran the GH checks and it was added https://github.com/vmware-tanzu/cluster-api-provider-bringyourownhost/pull/347#issuecomment-1020793029.

**Comment by anusha94** [comment]
> are you suggesting to define another make target for GH Actions workflows. I assumed we used make test-<component> targets for local dev/test.

I think what we have is fine. Its just the name that's misleading I suppose. It looks like we are depending on the target that generates codecov report as our source of testing.
Instead it is just running all the tests and also generating codecov along with it. To include webhook-test also in this target after we skip for apis package, we can probably set include for the webhook files using `--focus-file`.
OR now that we have a codecov.yaml, we can probably set the skip / include parameters in that file?

File: codecov.yml (hunk 1 above)
Comment (dharmjit) [reply codecov.yml:4]: Yes, it makes sense. I kept it only to test this.

**Comment by dharmjit** [comment]
> OR now that we have a codecov.yaml, we can probably set the skip / include parameters in that file?

Yes, I guess it would be easier to handle it in `codecov.yml` than at `ginkgo` but we could also look at keeping it consistent across tools. This we can pick in #344

**Review by anusha94 — APPROVED** [review]
lgtm

**Review by huchen2021 — APPROVED** [review]

**Pull request merged**